packages/calendar-input/src/date-input.tsx

```diff
@@ -0,0 +1,275 @@
+        )
+      }
+    } else {
+      onChange()
+      setInputValue(newValue)
+    }
+  }
+
+  const setSelectionRange = (start: number, end: number) => {
+    if (inputNode.current) {
+      inputNode.current.childNodes.forEach(node => {
+        if (node.nodeName === 'INPUT' && activeDatePart) {
+          ;(node as HTMLInputElement).focus()
+          ;(node as HTMLInputElement).setSelectionRange(start, end)
+        }
+      })
+    }
+  }
+
+  const focusNextPart = () => {
+    if (activeDatePart) {
+      const activeDatePartIndex = CalendarDate.format.indexOf(activeDatePart)
+      if (activeDatePartIndex + 1 < CalendarDate.format.length) {
+        setActiveDatePart(CalendarDate.format[activeDatePartIndex + 1])
+        setCursorPosition(
+          CalendarDate.getPartStartIndex(
+            CalendarDate.format[activeDatePartIndex + 1]
+          )
+        )
+      }
+    }
+  }
+  const focusPrevPart = () => {
+    if (activeDatePart) {
+      const activeDatePartIndex = CalendarDate.format.indexOf(activeDatePart)
+      if (activeDatePartIndex > 0) {
+        setActiveDatePart(CalendarDate.format[activeDatePartIndex - 1])
+        setCursorPosition(
+          CalendarDate.getPartStartIndex(
+            CalendarDate.format[activeDatePartIndex - 1]
+          )
+        )
+      }
+    }
+  }
+
+  const isStartCurrentPart = () => {
+    if (activeDatePart) {
+      return CalendarDate.getPartStartIndex(activeDatePart) === cursorPosition
+    }
+    return false
+  }
+  const isEndCurrentPart = () => {
+    if (activeDatePart) {
+      return CalendarDate.getPartEndIndex(activeDatePart) - 1 === cursorPosition
+    }
+    return false
+  }
+
+  const setPartValue = (part: DateTimePartsType, value) => {
+    let newValue = Number(value)
+    if (isNaN(newValue)) {
+      changeInputValue({...inputValue, [part]: value})
+    } else {
+      if (part === 'month' && newValue > 12) {
+        newValue = 12
+      } else if (part === 'day' && newValue > 31) {
+        newValue = 31
+      } else if (part === 'hours' && newValue > 23) {
+        newValue = 23
+      } else if (part === 'minutes' && newValue > 59) {
+        newValue = 59
+      }
+      changeInputValue({...inputValue, [part]: newValue})
+    }
+  }
+
+  const moveCursorRight = (count: number = 1) => {
+    for (let i = 0; i < count; i++) {
+      if (isEndCurrentPart()) {
+        focusNextPart()
+      } else if (
+        cursorPosition + 1 <
+        CalendarDate.getStringValueForParts(inputValue).length
+      ) {
+        setCursorPosition(cursorPosition + 1)
+      }
+    }
+  }
+
+  const moveCursorLeft = (count: number = 1) => {
+    for (let i = 0; i < count; i++) {
+      if (isStartCurrentPart()) {
+        focusPrevPart()
+      } else if (cursorPosition > 0) {
+        setCursorPosition(cursorPosition - 1)
+      }
+    }
+  }
+
+  const handleKeyDown = (e: React.KeyboardEvent<{}>) => {
+    e.stopPropagation()
+    if (e.keyCode === KEYS.DOWN) {
```

Comment:
`keyCode` is being deprecated in favor of `key`: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode